### PR TITLE
parser: Unify time unit string literal

### PR DESCRIPTION
**Pull request by shenli**
For timestampdiff() we do not unify time unit literal, so it is case sensitive. And in expression/types we always use upper case to compare time unit.

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

File: parser/parser_test.go [hunk 1]
```diff
@@ -1305,3 +1305,23 @@ func (s *testParserSuite) TestExplain(c *C) {
 	}
 	s.RunTest(c, table)
 }
+
+func (s *testParserSuite) TestTimestampDiffUnit(c *C) {
+	// Test case for timestampdiff unit.
+	// TimeUnit should be unified to upper case.
+	parser := New()
+	stmt, err := parser.Parse("SELECT TIMESTAMPDIFF(MONTH,'2003-02-01','2003-05-01'), TIMESTAMPDIFF(month,'2003-02-01','2003-05-01');;", "", "")
```
Comment (tiancaiamao) [inline parser/parser_test.go]: A redundant `;`

Comment (shenli) [inline parser/parser_test.go]: Good catch! PTAL

**Pull request merged**